partner_interest_group/models/res_partner.py

```diff
@@ -0,0 +1,7 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    interest_groups_ids = fields.Many2many("res.partner.interest.groups")
```

Comment:
Here, I would suggest to remove plural form to `groups` as the plural is indicated by the `ids` write:
```python
interest_group_ids = ...
```

Reply:
fixed